File: app/src/main/java/com/vincent/githubusersearch/network/ApiInterface.java
package com.vincent.githubusersearch.network;

import com.vincent.githubusersearch.model.ApiUrl;
import com.vincent.githubusersearch.model.ItemSearchResult;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiInterface {

    @GET(ApiUrl.API_GITHUB_USER)
    Call<ItemSearchResult> getGitHubUsers(@Query("q") String keyword, @Query("per_page") int perPage, @Query("page") int page,
                                          @Query("client_id") String clientId, @Query("client_secret") String clientSecret);

}

File: app/src/main/java/com/vincent/githubusersearch/network/ApiInterface.kt
package com.vincent.githubusersearch.network

import com.vincent.githubusersearch.model.ApiUrl
import com.vincent.githubusersearch.model.ItemSearchResult
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiInterface {

    @GET(ApiUrl.API_GITHUB_USER)
    fun getGitHubUsers(@Query("q") keyword: String?, @Query("per_page") perPage: Int, @Query("page") page: Int,
                       @Query("client_id") clientId: String, @Query("client_secret") clientSecret: String): Call<ItemSearchResult>
}